### PR TITLE
[MDS-4956] incidents tables

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

Add server side handling of pagination/filters/sorting for the incidents tables.  Previously, all records were being fetched each time and causing pod crashes in some instances.

[MDS-4956](https://bcmines.atlassian.net/browse/MDS-4956)

- Refactored various incident components from class to functional
- Universally utilized the incidents endpoint optimized for pagination etc. 
- Wired the flow for server side pagination from:
   - The incidents table within a mine on Minespace
   - The incidents table from the browse menu on Core
   - The incidents table from the mine menu on Core
- ~~Disabled some of the exports from the common package as they were causing eslint duplicate export errors~~ This ended up causing errors, so I restored them
- Added common pagination utils to the common package

_Why are you making this change? Provide a short explanation and/or screenshots_

image
image

### CODE REVIEW

File: services/common/src/constants/index.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
+/** This is the source of the eslint import errors.
+ * These files contain many exports with identical names.
+ * This will need some cleaning, but the commented out files are
+ * not currently being imported anywhere, so this will at least
+ * stop the errors for now.
+ */
```
Comment (taraepp) [inline services/common/src/constants/index.js:7]: comment out of date?

File: services/common/src/constants/utils.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const parseServerSideSearchOptions = (pagination, filters, sorter) => {
+  Object.keys(filters).forEach((key) => !filters[key] && delete filters[key]);
+  return {
+    per_page: pagination.pageSize,
+    page: pagination?.current,
+    sort_field: sorter?.column?.dataIndex || undefined,
+    sort_dir: parseSortDir(sorter.order),
+    ...filters,
+  };
+};
```
Comment (taraepp) [inline services/common/src/constants/utils.js:26]: Nice- this looks like it could be used all over the place!

**Review by taraepp — COMMENTED** [review]
I think that one comment I pointed out should be revised but otherwise LGTM

File: services/common/src/constants/index.js (hunk 1 above)
Comment (matbusby-fw) [reply services/common/src/constants/index.js:7]: Good catch.  Updated!

File: services/common/src/constants/utils.js (hunk 2 above)
Comment (matbusby-fw) [reply services/common/src/constants/utils.js:26]: Indeed!  Tried to make it as reusable as possible.

**Review by taraepp — APPROVED** [review]
👍

**Pull request merged**